packages/blade/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,523 @@
+  >
+> & {
+  activeIndicator: number;
+  totalSlides: number;
+  onIndicatorButtonClick: (index: number) => void;
+  onNextButtonClick: () => void;
+  onPreviousButtonClick: () => void;
+};
+
+const Controls = ({
+  showIndicators,
+  navigationButtonPosition,
+  activeIndicator,
+  totalSlides,
+  onIndicatorButtonClick,
+  onNextButtonClick,
+  onPreviousButtonClick,
+  indicatorVariant,
+  navigationButtonVariant,
+}: ControlsProp): React.ReactElement => {
+  // 1. buttons or indicators side by side on bottom
+  // 3. only indicators on bottom and buttons will be on side
+  const isNavButtonsOnBottom = navigationButtonPosition === 'bottom';
+  const case3 = showIndicators && navigationButtonPosition === 'side';
+
+  if (isNavButtonsOnBottom) {
+    return (
+      <Box marginTop="spacing.7" display="flex" alignItems="center" gap="spacing.4">
+        <NavigationButton
+          type="previous"
+          variant={navigationButtonVariant}
+          onClick={onPreviousButtonClick}
+        />
+        {showIndicators ? (
+          <Indicators
+            onIndicatorButtonClick={onIndicatorButtonClick}
+            activeIndex={activeIndicator}
+            totalItems={totalSlides}
+            variant={indicatorVariant}
+          />
+        ) : null}
+        <NavigationButton
+          onClick={onNextButtonClick}
+          type="next"
+          variant={navigationButtonVariant}
+        />
+      </Box>
+    );
+  }
+
+  if (case3) {
+    return (
+      <Box marginTop="spacing.7">
+        <Indicators
+          onIndicatorButtonClick={onIndicatorButtonClick}
+          activeIndex={activeIndicator}
+          totalItems={totalSlides}
+          variant={indicatorVariant}
+        />
+      </Box>
+    );
+  }
+
+  return <></>;
+};
+
+const CarouselContainer = styled(BaseBox)<{
+  showOverlay?: boolean;
+  scrollOverlayColor: CarouselProps['scrollOverlayColor'];
+  isScrollAtStart: boolean;
+  isScrollAtEnd: boolean;
+}>(({ theme, showOverlay, scrollOverlayColor, isScrollAtStart, isScrollAtEnd }) => {
+  const gradientStop1: string = getIn(theme.colors, scrollOverlayColor as string);
+  const gradientStop2 = 'hsla(0, 0%, 100%, 0)';
+
+  const overlayCommonStyle: CSSObject = {
+    content: "''",
+    position: 'absolute',
+    top: 0,
+    width: '100px',
+    height: '100%',
+    transition: '400ms ease',
+    transitionProperty: 'opacity',
+  };
+
+  return {
+    width: '100%',
+    overflowX: 'scroll',
+    display: 'flex',
+    flexWrap: 'nowrap',
+    scrollSnapType: 'x mandatory',
+    scrollSnapPointsY: `repeat(100%)`,
+    scrollBehavior: 'smooth',
+    msOverflowStyle: 'none' /* IE and Edge */,
+    scrollbarWidth: 'none' /* Firefox */,
+    /* Needed to work on iOS Safari */
+    webkitOverflowScrolling: 'touch',
+    msScrollSnapType: 'mandatory',
+    scrollSnapPointsX: 'repeat(100%)',
+    msScrollSnapPointsX: 'repeat(100%)',
```

Comment:
Check this: https://ishadeed.com/article/css-scroll-snap/

Actually only 2 lines of code is needed `scroll-snap-type, scroll-snap-align`, but other stuff is just alternatives for older browsers. (styled components won't auto prefix them because it's actually entirely different older spec that older browsers follow)